controllers/user/login.js

```diff
@@ -0,0 +1,48 @@
+const passport = require('passport');
+const { makeErrorList, makeUserLink } = require('./utils');
+
+// try to validate a user's login request
+// @TODO consider rate limiting?
```

Comment:
Suggestion: Depending on the level at which you'd want to do this at [nginx has pretty good in built rate-limiting](http://nginx.org/en/docs/http/ngx_http_limit_req_module.html) and AFAIK WAF can do this kind of thing too I'd probably err towards WAF > NGINX > Application in terms of preference.

Reply:
Yeah, this is perhaps a post-launch problem.